### PR TITLE
Add brotli support

### PR DESCRIPTION
**Pull request by jabley**
Brotli is a new open-source compression algorithm from Google which can
give more dense compression, while compression and decompression speeds
are roughly the same.

Brotli responses are requested by setting `Accept-Encoding: br`.

Transferring less data over the network will typically result in a
faster experience for users.

This change adds support for Brotli decompression of responses.

### CODE REVIEW

**Comment by iMacTia** [comment]
Hi @jabley, I'm really sorry your PR went ignored for so long!
I'm now in the process of catching up and this PR looks interesting but we have some issues with ruby < 2.0.
In order to get this merged in the next release, I'd need support for at least ruby 1.9.3. Is there any backport you can add to your PR to make tests passing on ruby 1.9.3 (please rebase with master to get rid of ruby 1.8.7 tests)

**Comment by jabley** [comment]
I'll see if the brotli gem will accept a pull request to support < Ruby 2.x.

**Comment by jabley** [comment]
https://github.com/miyucy/brotli/pull/23

If there are problems with that, what's your support policy on 1.9.3? It's been EOLed for quite a while.

**Comment by iMacTia** [comment]
The idea is that we keep supporting as many ruby versions as possible, depending on how this impact our work. Reality is that even though 1.9.3, 2.0 and 2.1 have been discontinued, there are still many "legacy" applications using them.

I understand `brotli` support is a nice-to-have feature, but it's hard to justify the loss of support for some older ruby version to provide a minor feature. That will prevent people using those rubies to update Faraday and get the latest features and fixes.

On the other hand, we can't keep supporting ancient ruby versions. That's why we'll definitely abandon 1.9.3 with the release of Faraday v1.0. I think a major release is the perfect time to decide which rubies we should support and which we shouldn't. My initial plan was to just drop 1.9.3, but given that 2.0 and 2.1 are both already EOL, we might think to push the threshold even further.

In short, I appreciate you trying to add support for 1.9.3 on brotli and I'll be happy to merge this in for release 0.13.0 if that happens. However, should that not be the case, we can simply move your PR against the v1.0 branch 👍

**Comment by jabley** [comment]
It is done. :shipit: 😀

**Comment by iMacTia** [comment]
Great work in maintaining support for ruby 1.9.3.
I really appreciate you having to update other gems as well to accomplish this.
Really happy to merge and mark for next release 👍
